### DISCUSSION TITLE
classes/class_animationtree

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#237] [User-contributed notes]
# classes/class_animationtree

Inherits: AnimationMixer< Node< Object A node used for advanced animation transitions in an AnimationPlayer. Description: A node used for advanced animation transitions in an AnimationPlayer. Note:...

https://docs.godotengine.org/en/latest/classes/class_animationtree.html

**Comment by SlavaSTiwari** [comment]
Please make an `is_finished` or  `is_playing()` method/signal like in the `Animatio Player`. since even tho the blend trees and state machine in the Animation Tree use anmations from an existig animation player.  `is_finished` or  `is_playing()` are not triggered for the Player when the Tree is used.

If i am wrong and there IS a way to check when an animation is finished in the tree, please let me know, thanks!

**Reply by tot121212** [reply]
So I was just having this issue and I believe I have figured it out. It appears the AnimationTree creates a duplicate of the AnimationPlayer instead of using it during runtime. All this time, I was trying to edit the AnimationPlayer itself but it turns out you can just access the AnimationTree as you would the AnimationPlayer and it just works. 
So any properties or signals that you would normally use from the animation player, just use them from the animation tree itself.

```Note: When linked with an AnimationPlayer, several properties and methods of the corresponding AnimationPlayer will not function as expected. Playback and transitions should be handled using only the AnimationTree and its constituent AnimationNode(s). The AnimationPlayer node should be used solely for adding, deleting, and editing animations.```

**Reply by CoffeePasta-cloud** [reply]
What @tot121212 said is working. Thanks a lot!

**Reply by leonardoraele** [reply]
`AnimationTree` is currently missing a lot of functionality about reacting to it. You can't be notified when state machine transitions from one state to the other, for example. You can't read what animation nodes are in use at a given time. You can't call methods from it, like you can using "call method" tracks. You can listen to `animation_started` and `animation_finished` signals emitted by the `AnimationTree` node, but no signal is emitted if the animations are being blended.

**Comment by itszygote** [comment]
If you are searching for an away to add more conditions to your transitions on a Animation state machine. 
https://www.youtube.com/watch?v=iosqXfNdJVw
This video helped me a lot. And a detail, this expression thing can be done with any node variable, including the ones of a script attached to the node. Basically you can use your own conditions and variables too!

**Comment by leonardoraele** [comment]
While an `AnimationTree` is active, the `AnimationPlayer` it controls remain stopped. i.e. `current_animation == &""`, `is_playing() == false`, `is_animation_active() == false`, and no signal will be emitted (from the `AnimationPlayer` node).
